### DISCUSSION TITLE
Plans for Apollo Client 3.0

### DISCUSSION THREAD

**Discussion by miller-productions** [dotansimha/graphql-code-generator#4014] [General] [answered]
Hi all,

Just started playing around with graphql-code-generator. Looks amazing.

Have run into a bit of a stumbling block however with the plugin typescript-react-apollo.

I'm using apollo client 3.0.

If I set `withHooks: true` in my codegen.yml config, graphql-code-generator produces an import statement such as...

`import * as ApolloReactHooks from "@apollo/react-hooks";`

Unfortunately, with this setup, you end up with an error `Module not found: Can't resolve 'apollo-client' in '<repo_path>/node_modules/@apollo/react-hooks/lib'`

It looks like it is due to the fact that in apollo client 3.0, hooks are included in the client package rather than in `@apollo/react-hooks`

See here for more details: https://github.com/apollographql/apollo-client/issues/5808#issuecomment-575921824

Q: are there plans for graphql-code-generator to support apollo client 3.0?

**Comment by ardatan** [comment, marked as answer]
https://graphql-code-generator.com/docs/generated-config/typescript-react-apollo#reactapolloversion-2--3-default-value-2
We already support Apollo Client v3.

**Reply by miller-productions** [reply]
Hi @ardatan, yes, my apologies. I discovered that earlier today also.

At first I didn't see that section but it's obvious now. Appreciate you getting back to me 😀

**Reply by miller-productions** [reply]
For anyone else reading this later, further details and discussion on it are available here...

https://github.com/dotansimha/graphql-code-generator/issues/3966#issuecomment-623003507
